src/Kiota.Builder/PathSegmenters/GoPathSegmenter.cs

```diff
@@ -85,7 +84,13 @@ public override string NormalizeFileName(CodeElement currentElement)
         return currentElement switch
         {
             CodeNamespace => "go",
-            _ => GetLastFileNameSegment(currentElement).ToSnakeCase().EscapeSuffix(specialFileNameSuffixes).ShortenFileName(100),
+            _ => GetLastFileNameSegment(currentElement)
+                .Replace("_", "_escaped_", StringComparison.OrdinalIgnoreCase)// at this point we have the guarantee that the element name is unique based on how child elements are held in Kiota.
```

Comment:
isn't that going to trigger when the file name is escaped?
e.g. `foo_windows_escaped.go` will become `foo_windows_escapedescaped.go` (or worse)

Reply:
Two things here.
- This escaping happens before escaping the file name. So, filename will remains as expected. (Existing filename tests are passing)
- `foo_windows_escaped` will become `foo_windows_escaped_escaped` if the original element(`CodeFile` or `CodeEnum`) is already called ``foo_windows_escaped``

Lemme see what I can do to improve the possibility of the second scenario here.

Reply:
I think we're good from a file name escaping perspective (go will only compile _windows.go files on windows, etc... this is why we had to escape the file names here)

From a code symbol escaping perspective, I'm a bit worried this will lead to double escaping in the file name. Maybe this should check the symbol name as well?

Reply:
The logic is now updated to cate for scenario 2. To avoid double escaping.

So, if an element has escaped to `foo_windows_escaped` the file name will still become `foo_windows_escaped` to avoid double escaping.

The symbole escaping logic for golang does not really escape symbols using an underscore from my investigation.
Escaping a symbol like `windows` becomes `windowsEscaped` which will then have the file name as `windows_escaped`.
With the current logic this should have no effect on such file symbols/files as shown in the updated tests.

Reply:
Thank you for making the changes!